go/engine/merkle_audit.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2019 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+// MerkleAudit runs a merkle tree audit in the background once in a while.
+// It verifies the skips of a randomly chosen merkle tree root, making
+// sure that the server is not tampering with the merkle trees.
+
+package engine
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+var (
+	errAuditOffline    = errors.New("Merkle audit failed to run due to the lack of connectivity.")
+	errAuditNoLastRoot = errors.New("Merkle audit failed to run due to not being able to get the last root.")
+)
+
+var MerkleAuditSettings = BackgroundTaskSettings{
+	Start:        5 * time.Minute,
+	StartStagger: 1 * time.Hour,
+	Interval:     6 * time.Hour,
+	Limit:        1 * time.Minute,
```

Comment:
should set `DisableInBotLite: true` once https://github.com/keybase/client/pull/15672 is merged (or i'll update that PR if this is merged first)